common/table.js

```diff
@@ -61,6 +61,7 @@
         // If it returns true then we should render the data
         // else we should reject the data
         function setSearchStates(scope, isBackground, searchTerm) {
+            // TODO: Wrap this function body in try/catch?
```

Comment:
because this is only used in the following `read()` function, I believe we have to wrap the body here.